content/blog/oauth2-for-hpe-greenlake-data-services-cloud-console.md

```diff
@@ -0,0 +1,101 @@
+---
+title: "Implementing OAuth 2 Flow for Data Services Cloud Console's Client
+  Application "
+date: 2021-10-14T12:46:19.488Z
+author: Ron Dharma
+authorimage: /img/face-portrait-small.jpg
+thumbnailimage: /img/dscc-icon-transparent.png
+tags:
+  - data-services-cloud-console
+---
+## HPE GreenLake API Security
+
+The major principle for public (Northbound) API to HPE GreenLake and Data Services Cloud Console is *secured interface that support authentication and authorization of the resource owner.* To support the previously mentioned principle, the HPE GreenLake implements the Oauth2 authorization granting the resource owner authorization rights without embedding the resource owner's user credential as in the API streams. This goal is achieved using the Oauth2 authorization and REST API, where every API call will embed the access token as part of the HTTP REST API URL stream using the keyword: *bearer*.
+
+At the introduction of this public API, HPE GreenLake Console supports the **Client Credential authentication grant** **type** (a.k.a. OAuth 2 client-credential authentication workflow.) This particular grant type allows the client application to authenticate using separate credentials (Client ID and Client Password) that is authorized inside the API Gateway menu using the HPE GreenLake User's Email and Password.
+
+**Some of the benefits of DSCC Client Credential OAuth authentication grant:**
+
+1. The authentication for Client does not involve the transmission of the HPE GreenLake user credentials.
+2. Changing the *Client Secret* or deleting the *Client ID* will not impact HPE GreenLake user credentials.
+3. According to OAuth 2.0 [https://tools.ietf.org/html/rfc6749#section-4.4](https://datatracker.ietf.org/doc/html/rfc6749#section-4.4), Client Credential grant type allows the client Application to authenticate by itself independent of user (no user intervention) which makes this grant type appropriate for machine-to-machine (M2M) application that can safely protect the registered client credentials (Confidential Clients) such as scripts, daemon, or services contained in a host. Please refer to this [](https://tools.ietf.org/html/rfc6749#section-2.1)<https://tools.ietf.org/html/rfc6749#section-2.1> for more information.
+4. Each client application uses different set of client ID and client secret to ensure secrecy and independency of each applications.
+
+### So, how do I implement HPE DSCC API in my client application?
+
+After an user obtains the Client ID and Client Password as shown in below picture. The flow to get the Client ID and Client Secret from this menu is detailed in my blog titled **Using API Gateway to Data Services Cloud Console.**
+
+![image of the client id and client secret](/img/credentials-created-client.png "Client Credentials")
```

Comment:
Something to think about... if we ever use a Preview to promote this blog post (which WILL happen, whether it's in Yammer or Twitter or Teams), the Preview pulls up the first image in the post. I don't think this would be the image you'd want associated with the promotion of the blog post - I don't think it gives enough information that's associated with its title. Can you insert another image higher up in the post that would?

Reply:
All set, I inserted the diagram for the flow of greenlakes access token acquisition which provides good rehash from the other blog.